src/Microsoft.AspNetCore.Mvc.Abstractions/Formatters/InputFormatterExceptionModelStatePolicy.cs

```diff
@@ -1,11 +1,51 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+
 namespace Microsoft.AspNetCore.Mvc.Formatters
 {
+    /// <summary>
+    /// Defines the set of policies that determine how the model binding system interprets exceptions
+    /// thrown by an <see cref="IInputFormatter"/>. Applications should set 
+    /// <c>MvcOptions.InputFormatterExceptionModelStatePolicy</c> to configure this setting.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// An <see cref="IInputFormatter"/> could throw an exception for several reasons including:
+    /// <list type="bullet">
+    /// <item><description>malformed input</description></item>
+    /// <item><description>client disconnect or other I/O problem</description></item>
+    /// <item><description>
+    /// application configuration problems such as <see cref="TypeLoadException"/>
+    /// </description></item>
+    /// </list>
+    /// </para>
+    /// <para>
+    /// The policy associated with <see cref="InputFormatterExceptionModelStatePolicy.AllExceptions"/> treats
+    /// all such categories of problems as model state errors, and usually will be reported to the client as
+    /// an HTTP 400. This was the only policy supported by model binding in ASP.NET Core MVC 1.0, 1.1, and 2.0
+    /// and is stil the default for historical reasons. 
+    /// </para>
+    /// <para>
+    /// The policy associated with <see cref="InputFormatterExceptionModelStatePolicy.MalformedInputExceptions"/>
+    /// treats only <see cref="InputFormatterException"/> and subclasses as model state errors. This means that
+    /// exceptions that are not related to the content of the HTTP request (such as a disconnect) will be rethrown.
```

Comment:
Would you add this at the end: `, which by default would cause an HTTP 500 response, unless there is exception-handling middleware enabled.`? Or is that too much?

Reply:
👍